subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -129,7 +129,7 @@ private void attach(ProgressOperation operation) {
         }
 
         // No parent? Try to use a new label
-        if (association == null && !unusedProgressLabels.isEmpty()) {
+        if (!unusedProgressLabels.isEmpty()) {
```

Comment:
The logic in that class, it just so happened that `association` is _always_ null at this point.